Unable to install `@langchain/core` when cloning fresh repo with newest changes. `tsdown` issue

### Checked other resources

- [x] This is a bug, not a usage question. For questions, please use the LangChain Forum (https://forum.langchain.com/).
- [x] I added a very descriptive title to this issue.
- [x] I searched the LangChain.js documentation with the integrated search.
- [x] I used the GitHub search to find a similar question and didn't find it.
- [x] I am sure that this is a bug in LangChain.js rather than my code.
- [x] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

Run following commands:
```
git clone --branch main --single-branch --depth 100 https://github.com/langchain-ai/langchainjs.git
```
```
cd langchainjs
```
```
pnpm install
```

```
pnpm --filter @langchain/core build 
```

### Error Message and Stack Trace (if applicable)

```
(base) filipzmijewski@Mac langchainjs % pnpm install
Scope: all 48 workspace projects
Lockfile is up to date, resolution step is skipped
Packages: +2575
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Progress: resolved 2575, reused 2550, downloaded 0, added 2575, done

devDependencies:
+ @changesets/changelog-github 0.5.2
+ @changesets/cli 2.29.8
+ @langchain/build 0.1.1 <- internal/build
+ @langchain/tsconfig 0.0.1 <- internal/tsconfig
+ @tsconfig/recommended 1.0.13
+ @types/jest 30.0.0
+ @types/semver 7.7.1
+ commander 11.1.0
+ dotenv 16.6.1
+ lint-staged 16.2.7
+ prettier 3.8.1
+ semver 7.7.4
+ tsdown 0.20.3
+ tsx 4.21.0
+ turbo 2.8.7
+ typescript 5.9.3

╭ Warning ─────────────────────────────────────────────────────────────────────╮
│                                                                              │
│   Ignored build scripts: @firebase/util, @prisma/client, @prisma/engines,    │
│   @swc/core, better-sqlite3, bufferutil, closevector-hnswlib-node,           │
│   couchbase, cpu-features, esbuild, faiss-node, hnswlib-node,                │
│   node-llama-cpp, onnxruntime-node, prisma, protobufjs, puppeteer, sharp,    │
│   sqlite3, ssh2, tesseract.js, unrs-resolver, usearch.                       │
│   Run "pnpm approve-builds" to pick which dependencies should be allowed     │
│   to run scripts.                                                            │
│                                                                              │
╰──────────────────────────────────────────────────────────────────────────────╯

Done in 14.4s using pnpm v10.14.0
(base) filipzmijewski@Mac langchainjs % pnpm --filter @langchain/core build

> @langchain/core@1.1.24 build /Users/filipzmijewski/Desktop/langchainjs/libs/langchain-core
> turbo build:compile --filter @langchain/core --output-logs new-only

╭──────────────────────────────────────────────────────────────────────────╮
│                                                                          │
│                     Update available v2.8.7 ≫ v2.8.9                     │
│    Changelog: https://github.com/vercel/turborepo/releases/tag/v2.8.9    │
│          Run "pnpm dlx @turbo/codemod@latest update" to update           │
│                                                                          │
│          Follow @turborepo for updates: https://x.com/turborepo          │
╰──────────────────────────────────────────────────────────────────────────╯
• turbo 2.8.7
• Packages in scope: @langchain/core
• Running build:compile in 1 packages
• Remote caching disabled
@langchain/core:build:compile: cache miss, executing 4851cf42b5c2a541
@langchain/core:build:compile: 
@langchain/core:build:compile: 
@langchain/core:build:compile: > @langchain/core@1.1.24 build:compile /Users/filipzmijewski/Desktop/langchainjs/libs/langchain-core
@langchain/core:build:compile: > tsdown
@langchain/core:build:compile: 
@langchain/core:build:compile: ℹ tsdown v0.20.3 powered by rolldown v1.0.0-rc.3
@langchain/core:build:compile: 
@langchain/core:build:compile:  ERROR  Error: [unrun] Import failed (code length: 2539): Unknown file extension ".ts" for /Users/filipzmijewski/Desktop/langchainjs/internal/build/src/index.ts
@langchain/core:build:compile:     at unrun (file:///Users/filipzmijewski/Desktop/langchainjs/node_modules/.pnpm/unrun@0.2.27_synckit@0.11.12/node_modules/unrun/dist/src-CS8hJStK.mjs:865:9)
@langchain/core:build:compile:     at async unrunImport (file:///Users/filipzmijewski/Desktop/langchainjs/node_modules/.pnpm/tsdown@0.20.3_@arethetypeswrong+core@0.18.2_@typescript+native-preview@7.0.0-dev.202602_b7c44a56216d612cdc3bfdd7e2ee3202/node_modules/tsdown/dist/options-DfwuH01N.mjs:497:21)
@langchain/core:build:compile:     at async iterateAsync (file:///Users/filipzmijewski/Desktop/langchainjs/node_modules/.pnpm/quansync@1.0.0/node_modules/quansync/dist/src-C2Pm6gXo.js:58:28)
@langchain/core:build:compile:     at async iterateAsync (file:///Users/filipzmijewski/Desktop/langchainjs/node_modules/.pnpm/quansync@1.0.0/node_modules/quansync/dist/src-C2Pm6gXo.js:58:28)
@langchain/core:build:compile:     at async iterateAsync (file:///Users/filipzmijewski/Desktop/langchainjs/node_modules/.pnpm/quansync@1.0.0/node_modules/quansync/dist/src-C2Pm6gXo.js:58:28)
@langchain/core:build:compile:     at async loadConfigFile (file:///Users/filipzmijewski/Desktop/langchainjs/node_modules/.pnpm/tsdown@0.20.3_@arethetypeswrong+core@0.18.2_@typescript+native-preview@7.0.0-dev.202602_b7c44a56216d612cdc3bfdd7e2ee3202/node_modules/tsdown/dist/options-DfwuH01N.mjs:418:19)
@langchain/core:build:compile:     at async resolveConfig (file:///Users/filipzmijewski/Desktop/langchainjs/node_modules/.pnpm/tsdown@0.20.3_@arethetypeswrong+core@0.18.2_@typescript+native-preview@7.0.0-dev.202602_b7c44a56216d612cdc3bfdd7e2ee3202/node_modules/tsdown/dist/build-C5RQkVSf.mjs:90:41)
@langchain/core:build:compile:     at async build$1 (file:///Users/filipzmijewski/Desktop/langchainjs/node_modules/.pnpm/tsdown@0.20.3_@arethetypeswrong+core@0.18.2_@typescript+native-preview@7.0.0-dev.202602_b7c44a56216d612cdc3bfdd7e2ee3202/node_modules/tsdown/dist/build-C5RQkVSf.mjs:784:42)
@langchain/core:build:compile:     at async CAC.<anonymous> (file:///Users/filipzmijewski/Desktop/langchainjs/node_modules/.pnpm/tsdown@0.20.3_@arethetypeswrong+core@0.18.2_@typescript+native-preview@7.0.0-dev.202602_b7c44a56216d612cdc3bfdd7e2ee3202/node_modules/tsdown/dist/run.mjs:23:2)
@langchain/core:build:compile:     at async runCLI (file:///Users/filipzmijewski/Desktop/langchainjs/node_modules/.pnpm/tsdown@0.20.3_@arethetypeswrong+core@0.18.2_@typescript+native-preview@7.0.0-dev.202602_b7c44a56216d612cdc3bfdd7e2ee3202/node_modules/tsdown/dist/run.mjs:47:3)
@langchain/core:build:compile: 
@langchain/core:build:compile:  ELIFECYCLE  Command failed with exit code 1.
@langchain/core:build:compile: ERROR: command finished with error: command (/Users/filipzmijewski/Desktop/langchainjs/libs/langchain-core) /Users/filipzmijewski/Library/pnpm/.tools/@pnpm+macos-arm64/10.14.0/bin/pnpm run build:compile exited (1)
@langchain/core#build:compile: command (/Users/filipzmijewski/Desktop/langchainjs/libs/langchain-core) /Users/filipzmijewski/Library/pnpm/.tools/@pnpm+macos-arm64/10.14.0/bin/pnpm run build:compile exited (1)

 Tasks:    0 successful, 1 total
Cached:    0 cached, 1 total
  Time:    2.516s 
Failed:    @langchain/core#build:compile

 ERROR  run failed: command  exited (1)
/Users/filipzmijewski/Desktop/langchainjs/libs/langchain-core:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @langchain/core@1.1.24 build: `turbo build:compile --filter @langchain/core --output-logs new-only`
Exit status 1
```

### Description

As explained before running above commands fail the process, it worked before for sure.
When downgrading tsdown back to 0.15.* form 0.20.* it works again so I would assume the issue is somewhere here.
This works in my old environment where I have previously installed `@langchain/core`.

### System Info

```
(base) filipzmijewski@Mac langchainjs % pnpm info langchain

langchain@1.2.24 | MIT | deps: 5 | versions: 387
Typescript bindings for langchain
https://github.com/langchain-ai/langchainjs/tree/main/libs/langchain/

keywords: llm, ai, gpt3, chain, prompt, prompt engineering, chatgpt, machine learning, ml, openai, embeddings, vectorstores

dist
.tarball: https://registry.npmjs.org/langchain/-/langchain-1.2.24.tgz
.shasum: ed6e466c15a9396e49dece045acb1596a8048d5a
.integrity: sha512-NpaOmDZ4dP16sLkY+Y49Q9mrV2fdTy81t+yTw7f3K7/zZtvuQk6IH/bIzfy3bdOD8TERethvR2mOmgueRDbZBw==
.unpackedSize: 3.4 MB

dependencies:
@langchain/langgraph-checkpoint: ^1.0.0 @langchain/langgraph: ^1.1.2            langsmith: >=0.5.0 <1.0.0               uuid: ^10.0.0                           zod: ^3.25.76 || ^4                     

maintainers:
- christian-bromann <mail@bromann.dev>
- nfcampos <nuno@boringbits.io>
- jacoblee93 <jacoblee93@gmail.com>
- andrewnguonly <andrewnguonly@gmail.com>
- davidduong <david@duong.cz>
- hntrl <hunter@hntrl.io>
- hwchase17 <hw.chase.17@gmail.com>
- basproul <braceasproul@gmail.com>

dist-tags:
alpha: 1.0.0-alpha.9                       dev: 1.2.9                                 latest: 1.2.24                             next: 1.0.0-alpha.9                        tag-for-publishing-older-releases: 0.2.20  

published 3 days ago by GitHub Actions <npm-oidc-no-reply@github.com>
(base) filipzmijewski@Mac langchainjs % node -v            
v20.19.2
(base) filipzmijewski@Mac langchainjs % pnpm -v
10.14.0
```

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -182,6 +182,13 @@ To get started, install the dependencies for the project from the root:
 pnpm install
 ```
 
+Make sure you have [`nvm`](https://github.com/nvm-sh/nvm?tab=readme-ov-file#installing-and-updating) installed and setup:
+
+```sh
+# switch to recommended Node.js version
+nvm use
+```
+
 Then, build the core package (required before working on other packages):
 
 ```bash

